### PR TITLE
Add JOSE partnership info

### PR DESCRIPTION
**Pull request by tobyhodges**
* Add info about JOSE reviews to README
* Add option for JOSE submission to author checklist & associated issue template
* Add some steps for editors to take after a lesson is accepted

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,10 +77,43 @@ A lesson is particularly well-suited to The Carpentries Lab if:
...
+  the accompanying paper in your lesson repository will be reviewed separately
+  by their team (see below).
 
 _This process is inspired by the software review from [rOpenSci][ropensci]_
 
+
+## Joint Review with JOSE
+
+When submitting their lesson for review in The Carpentries Lab,
+authors will have the option to also submit to the review process for
```
Comment (fmichonneau) [inline README.md:94]: should we specify here that the review with JOSE happens after the review with Carpentries Lab?
I'm assuming that, ideally, the JOSE reviewers and editors would want to see the types of comments/feedback and improvements made to the lesson following the carpentries lab review.

Comment (MCMaurer) [reply README.md:94]: Great point- I also wonder what would happen if a lesson was rejected from inclusion in the Carpentries Lab. Would that mean they can still submit to JOSE? Or are we saying that if you choose to submit via the Incubator -> Lab pipeline, acceptance to the Lab is a prerequisite for JOSE submission?

**Review by MCMaurer — COMMENTED** [review]
In line with @fmichonneau's comments, I think it would be good to clarify the role that reviewers play in the `paper.md` file, and the sequence of events for JOSE submission. Maybe we say the `paper.md` is an optional item on the reviewers checklist?

I also think we want to make it clear that JOSE submission happens after the Lab review process. If the lesson is accepted to the Lab (even after rounds of revision), the Lab Editor will assist with the next steps with JOSE. If the lesson is not accepted to the Lab, this will not happen, but we should make it clear what the authors can do at that point re: JOSE.

**Comment by tobyhodges** [comment]
I've added the question of pathways through review/what to do about reviews that stall/are not accepted to the Lab to the agenda for our next call.

I'm going to merge this now because it is important to get the basic info included ASAP - we can come back and add more details and clarifications later.

**Pull request merged**